pkg/statedb: panic in statedb.(*DeleteTracker[...]).Close

```
level=info msg=Invoked duration="253.093µs" function="statedb.TestDerive.Derive[...].func7 (pkg/statedb/derive.go:54)" subsys=hive
level=info msg=Starting subsys=hive
level=info msg="Start hook executed" duration="13.916µs" function="hive.New.func1.2 (pkg/hive/hive.go:104)" subsys=hive
level=info msg="Start hook executed" duration="1.813µs" function="*cell.reporterHooks.Start" subsys=hive
level=info msg="Start hook executed" duration="1.663µs" function="*statedb.DB.Start" subsys=hive
level=info msg="Start hook executed" duration="8.637µs" function="*job.group.Start" subsys=hive
level=info msg=Started duration="126.947µs" subsys=hive
level=info msg=Stopping subsys=hive
level=info msg="Stop hook executed" duration="46.136µs" function="*job.group.Stop" subsys=hive
level=info msg="Stop hook executed" duration="6.833µs" function="*statedb.DB.Stop" subsys=hive
level=info msg="Stop hook executed" duration="50.414µs" function="*cell.reporterHooks.Stop" subsys=hive
level=info msg="Stop hook executed" duration="1.623µs" function="hive.New.func1.3 (pkg/hive/hive.go:111)" subsys=hive
panic: send on closed channel

goroutine 106 [running]:
github.com/cilium/cilium/pkg/statedb.(*DeleteTracker[...]).Close(0x0)
	/host/pkg/statedb/deletetracker.go:76 +0x21e
github.com/cilium/cilium/pkg/statedb.(*observable[...]).Observe.func1()
	/host/pkg/statedb/observable.go:56 +0x3d4
created by github.com/cilium/cilium/pkg/statedb.(*observable[...]).Observe in goroutine 8
	/host/pkg/statedb/observable.go:33 +0xf7
FAIL	github.com/cilium/cilium/pkg/statedb	0.081s
```

https://github.com/cilium/cilium/actions/runs/7929011074/job/21648439532#step:16:647

--- a/pkg/statedb/db.go
+++ b/pkg/statedb/db.go
@@ -234,7 +234,6 @@ func (db *DB) Start(cell.HookContext) error {
 }
 
 func (db *DB) Stop(stopCtx cell.HookContext) error {
-	close(db.gcTrigger)
 	db.cancel()
 	select {
 	case <-stopCtx.Done():

--- a/pkg/statedb/graveyard.go
+++ b/pkg/statedb/graveyard.go
@@ -24,7 +24,13 @@ func graveyardWorker(db *DB, ctx context.Context, gcRateLimitInterval time.Durat
 	defer limiter.Stop()
 	defer close(db.gcExited)
 
-	for range db.gcTrigger {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-db.gcTrigger:
+		}
+
 		// Throttle garbage collection.
 		if err := limiter.Wait(ctx); err != nil {
 			return

